### ISSUE TITLE
Getting predictions from trees

### ISSUE THREAD

**Issue by Pibborn** [intuit/Decision-Trees-over-FHE#2]
Hi,

could you describe how to get **label values** as output from trained trees and forests? The example script seems to be doing the following:

1. Loading `../data/default_data.csv` as a training set
2. Creating a random instance as a test set of size 1
3. Using an evaluation server `eval_server`'s `execute(...)` function to get a prediction
4. Decrypting the prediction via `client.decrypt_prediction`

However, the output on stdout for this scenario with the default parameters given in the repo is, perplexingly, a vector with 16384 entries! Only the first two entries seem to be non-zero, however.

Simply put, how does one obtain a label prediction instead? In the given example, this should be either 0 or 1.

Thank you in advance!

**Comment by margar8** [comment]
Hi Pibborn,

We describe in ALGORITHM1 in the paper, the prediction algorithm. In our prediction algorithm the leaf is holding the
1-hot encoding of its label. The final prediction is a _weighted sum_ of the leaves (i.e., of these 1-hot encoded labels). To get the final prediction as a single label there are several possibilities: 
1. take max: the predicted label is the one with the max weight. 
2. use the weights to define label probability proportional to its relative weight and pick randomly a label where each label has the probability we defined based on the weight it received.

Best,
Margarita
